src/lib/otp_lib/capped_supervisor.ml

```diff
@@ -0,0 +1,46 @@
+open Core_kernel
+open Async_kernel
+open Pipe_lib
+open Strict_pipe
+
+type 'job t =
+  { job_writer: ('job, drop_head buffered, unit) Writer.t
+  ; f: 'job -> unit Deferred.t }
+
+let create ?(buffer_capacity = 10) ~job_capacity f =
+  let job_reader, job_writer =
+    Strict_pipe.create
+      (Buffered (`Capacity buffer_capacity, `Overflow Drop_head))
+  in
+  let active_jobs = ref 0 in
+  let pending_jobs = ref [] in
+  let bvar = Bvar.create () in
```

Comment:
Can you name this `bvar` into something else? It's a bit unclear what it does at first glance.